tests/test_spans_api.py

```diff
@@ -0,0 +1,306 @@
+                                                ),
+                                                "trace_id": self.trace_id,
+                                                "span_id": str(int(span_id, 16) + 2),
+                                                "data": {},
+                                                "op": "db",
+                                                "hash": "c" * 16,
+                                                "exclusive_time": 0.4567,
+                                            },
+                                        ],
+                                    },
+                                },
+                            ),
+                            KafkaMessageMetadata(0, 0, self.base_time),
+                        )
+                    )
+                    if processed:
+                        events.append(processed)
+        write_processed_messages(self.storage, events)
+
+    def _post_query(
+        self,
+        query: str,
+        turbo: bool = False,
+        consistent: bool = True,
+        debug: bool = True,
+    ) -> Any:
+        return self.app.post(
+            "/spans/snql",
+            data=json.dumps(
+                {
+                    "query": query,
+                    "turbo": False,
+                    "consistent": True,
+                    "debug": True,
+                }
+            ),
+            headers={"referer": "test"},
+        )
+
+    def test_count_for_project(self) -> None:
+        """
+        Test total counts are correct
+        """
+        from_date = (self.base_time - self.skew).isoformat()
+        to_date = (self.base_time + self.skew).isoformat()
+        response = self._post_query(
+            f"""MATCH (spans)
+                SELECT count() AS aggregate
+                WHERE project_id = 1
+                AND end_timestamp >= toDateTime('{from_date}')
+                AND end_timestamp < toDateTime('{to_date}')
+            """
+        )
+        data = json.loads(response.data)
+        assert response.status_code == 200, response.data
+        assert data["data"][0]["aggregate"] > 10, data
+
+        # No count for project 2 even though data was being sent to the processor because the
+        # allowlist does not allow the project
+        from_date = (self.base_time - self.skew).isoformat()
+        to_date = (self.base_time + self.skew).isoformat()
+        response = self._post_query(
+            f"""MATCH (spans)
+                SELECT count() AS aggregate
+                WHERE project_id = 2
+                AND end_timestamp >= toDateTime('{from_date}')
+                AND end_timestamp < toDateTime('{to_date}')
+            """
+        )
+        data = json.loads(response.data)
+        assert response.status_code == 200, response.data
+        assert data["data"][0]["aggregate"] == 0, data
+
+    def test_get_group_sorted_by_exclusive_time(self) -> None:
```

Comment:
This is the query which has been provided by product in https://github.com/getsentry/snuba/pull/4089#issuecomment-1526118481